src/Handler/StreamHandler.php

```diff
@@ -308,9 +308,17 @@ function () use ($context, $params) {
         );
 
         return $this->createResource(
-            function () use ($request, &$http_response_header, $context) {
+            function () use ($request, &$http_response_header, $context, $options) {
                 $resource = fopen((string) $request->getUri()->withFragment(''), 'r', null, $context);
                 $this->lastHeaders = $http_response_header;
+
+                if (isset($options['read_timeout'])) {
```

Comment:
Other uses of the $options array in this class don't make use of RequestOptions constants, so I tried to keep it consistent.